src/components/GridTable.vue

```diff
@@ -30,62 +32,54 @@
   </table>
 </template>
 <script setup lang="ts">
-/* eslint-enable @typescript-eslint/no-explicit-any */
 import { ref } from "vue";
+import stach from "../stach-sdk/stach";
 
-type CellDetails = {
-  groupLevel: number
-}
-type HeaderCellDetails = {
-  rowspan: number,
-  colspan: number,
-  columnIndex: number,
-  source: string
-}
+type Row = stach.factset.protobuf.stach.v2.RowOrganizedPackage.IRow;
+type IListValue = stach.google.protobuf.IListValue | null
+type IRow = stach.factset.protobuf.stach.v2.RowOrganizedPackage.IRow[] | null | undefined
+let table = ref<IRow>();
 
-type Row = {
-  cells: string[],
-  headerCellDetails?: HeaderCellDetails[] ,
-  rowType?: string,
-  cellDetails?: CellDetails[],
-}
+// fetch data from the server
+fetch("http://localhost:3000/data")
+  .then((response) => response.json())
+  .then((data) => {
+    const pkg =
+      stach.factset.protobuf.stach.v2.RowOrganizedPackage.create(data);
+    table.value = pkg.tables.main.data?.rows;
 
-const isHeader = (row: Row): boolean => {
-  return row.rowType === "Header";
-};
-const isHidden = (): boolean => {
-  return false;
+  });
+
+// isHeader checks if the row is a header row
+const isHeader = (row?: Row): boolean => {
+  return row?.rowType === ("Header" as Row);
```

Comment:
I am suspicious about `("Header" as Row)` here since `"Header"` should be a `RowType` not `Row`